psqlgraph/mocks.py

```diff
@@ -93,16 +93,30 @@ def validate_value(self, value):
         return isinstance(value, bool)
 
 
+class ArrayRand(Randomizer):
+    def random_value(self, override=None):
+        if self.validate_value(override):
+            return override
+
+        return []
+
+    def validate_value(self, value):
+        return isinstance(value, list)
```

Comment:
The values an array can take are possibly an enum, validating just the type will still let bad data through. In the TypeFactory, you can check the type of the array by looking at `[items]` key.

Reply:
So just want to make sure I follow you. You are saying that a array property can have an enum constraint correct?

Reply:
Yes, in fact the only valid use case is currently an enum

Reply:
So this should be all updated. Let me know if this wasn't what you were looking for.

Reply:
I updated this to generate a random length array (up to 5 elements) The validate_value though doesn't throw an exception in any other randomizer and I don't think its the best to introduce it here. However I did update it to return false if any of the values in the override list are bad to reflect your desire there.